src/HallPIC.jl

```diff
@@ -455,4 +457,107 @@ function boltzmann_electric_field_and_potential!(E, phi, n_e, T_e, grid)
     return E, phi
 end
 
+#======================================================
+Reactions definitions
+======================================================#
+
+struct ReactingSpecies
+    name::Symbol
+    coefficient::Int8 
+end
+struct Reaction{T<:AbstractFloat}
+    reactant::ReactingSpecies
+    products::Vector{ReactingSpecies}
+    threshold_energy::T
+    energies::Vector{T}
+    rate::Vector{T}
+    delta_n::Vector{T}
+
+end
+
+
+
+function reaction_reduction(grid::Grid, reaction::Reaction{T}, electron_properties::SpeciesProperties{T}, fluid_properties::SpeciesProperties{T}, reactant::ParticleContainer{T},  dt) where T
+
+    #for each (non-ghost) cell, calculate the delta n
+    for i in 2:length(grid.cell_centers)-1
+        #calculate the number of particles produced
+        rate_idx = findfirst(electron_properties.temp[i] .<= reaction.energies)
```

Comment:
you can use searchsortedfirst(reaction.energies, electron_properties) to avoid allocating memory